client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -519,25 +537,25 @@ private void setHeaderParams(RequestBuilder request, Method method, Object[] arg
         }
     }
 
-    protected Mono<Object> deserialize(Method method, String string) {
+    protected Flux<Object> deserialize(Method method, String string) {
         if (string == null || string.isEmpty()) {
-            return Mono.empty();
+            return Flux.empty();
         }
         Type type = ReflectionUtil.getTypeOfObservable(method);
 
         if (Void.class.equals(type)) {
-            return Mono.empty();
+            return Flux.empty();
         }
 
-        if (String.class.equals(type) && !string.startsWith(QUOTE) && !"null".equalsIgnoreCase(string)) {   
-            return just(string);
+        if (String.class.equals(type) && !string.startsWith(QUOTE) && !"null".equalsIgnoreCase(string)) {
+            return Flux.just(string);
         }
 
         try {
             JavaType     javaType = TypeFactory.defaultInstance().constructType(type);
             ObjectReader reader   = objectMapper.readerFor(javaType);
-            Object       value    = reader.readValue(string);
-            return Mono.justOrEmpty(value);
+            List<Object> value    = reader.readValues(string).readAll();
+            return Flux.fromArray(value.toArray());
```

Comment:
Flux.fromIterable could be used to avoid putting it into an array

But Even nicer would have been to avoid the reader.readValues(string.readAll()) as a blocking call outside the observbable as well when we are at it.

What do you think about:

return Flux.fromStream(() -> {
            try {
                return reader.readValues(string).readAll().stream();
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        });